pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -78,13 +84,53 @@ func (r *PullSecretReconciler) Reconcile(request ctrl.Request) (ctrl.Result, err
 			ps.Data = map[string][]byte{}
 		}
 
-		// validate
-		if !json.Valid(ps.Data[v1.DockerConfigJsonKey]) {
+		keyCondition := &status.Condition{
+			Type:   aro.RedHatKeyPresent,
+			Status: v1.ConditionFalse,
+			Reason: "CheckFailed",
+		}
+		samplesCondition := &status.Condition{
+			Type:   aro.SamplesOperatorModified,
+			Status: v1.ConditionFalse,
+			Reason: "RedHatKeyNotPresent",
+		}
+
+		// parse keys and validate JSON
+		parsedKeys, err := r.parseRHRegistryKeys(ps)
+		foundKeys := []string{}
+		if err != nil {
 			r.log.Info("pull secret is not valid json - recreating")
 			delete(ps.Data, v1.DockerConfigJsonKey)
+			keyCondition.Reason = "CheckDone"
+			keyCondition.Message = "Cannot parse secret data, recreating."
+		} else {
+			foundKeys = r.checkRHRegistryKeys(parsedKeys)
+			keyCondition = r.updateRHKeysCondition(foundKeys)
+		}
```

Comment:
This looks not clean. In one if you set condition fields directly, in other you have a helper function to do so. If we remove second key from the map this should simplify, and we should be able to reuse helper function and make it cleaner